### PR TITLE
refactor: [RND-542] Move trial debugging logic to server component

### PR DESCRIPTION
**Pull request by Youakeem**
## Justify why they are needed
We are now getting the form data directly instead of an endpoint body so the helpers needed to be updated accordingly.

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by Youakeem** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4068?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4068** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4068?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4063** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4063?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Youakeem and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/app/debugger/trial/actions.ts [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
-  })
+  const partner = getPartner(formData)
+  const data = getTrialData(formData)
+
+  let destination: string
+
+  try {
+    const { fullUrl } = await createTrial(partner, data)
+
+    const nextURL = new URL(fullUrl)
```
Comment (alebedev) [inline apps/store/src/app/debugger/trial/actions.ts:18]: What's easier - copying this part, or copying pathname and query params to `new URL(ORIGIN_URL)` ?

Comment (Youakeem) [reply apps/store/src/app/debugger/trial/actions.ts:18]: For me, it's easy to understand what's happening here at a glance. Don't think there is a huge difference with either approaches. Happy to change if you see a need.

**Review by alebedev — APPROVED** [review]

**Review by guilhermespopolin — APPROVED** [review]
🚀

**Pull request merged**